paper-autocomplete.js

```diff
@@ -652,23 +648,18 @@ Polymer({
    * @returns {Boolean}
    */
 	validate() {
-		return this.$.autocompleteInput.validate();
-	},
-
-	/**
-   * Clears the current input
-   */
-	clear() {
-		this._value = '';
-		this._text = '';
-		this._clear();
+		const valid = this.$.autocompleteInput.validate();
+		if (this.restricted && valid) {
```

Comment:
I don't understand how this works... How does this code ensure that the input value is one of the available options?

Reply:
`this.value` will always be `null`, `''` or one of values from options.
`this.value` is set to `''` or `null` when you type anything if it doesn't match.